File: src/asar-tests/test-init.cs
using System;
using System.IO;
using AsarCLR;

public static class Test
{
	public static int Main(string[] args){
		if(args.Length != 1)
		{
			Console.WriteLine("Usage: test-???.exe [path_to_tests_directory]");
			return 1;
		}

		if(!Asar.init())
		{
			Console.WriteLine("Unable to Initialize");
			return 1;
		}

		Asar.close();

		if(!Asar.init())
		{
			Console.WriteLine("Unable to Reinitialize");
			return 1;
		}

		Asar.close();

		return 0;
	}
}

File: src/asar-tests/test-init.py
import asar
from os import path
import sys

if len(sys.argv) != 2:
	print("Usage: test-???.py [path_to_tests_directory]")
	sys.exit(1)

if not asar.init():
	print("Unable to initialize Asar")
	sys.exit(1)
asar.close()

if not asar.init():
	print("Unable to re-initialize Asar")
	sys.exit(1)
asar.close()
